Testnet/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -36,148 +34,175 @@ public struct ApprovalForAllLog
 
     /// <summary>
     /// Sets the owner to operator permission.
     /// </summary>
     /// <param name="owner">The owner address of the NFT.</param>
     /// <param name="operatorAddress">>Address to add to the set of authorized operators.</param>
     /// <param name="value">The permission value.</param>
-    private void SetOwnerToOperator(Address owner, Address operatorAddress, bool value)
+    private void SetOwnerToOperator(Address owner, Address operatorAddress, bool value) => State.SetBool($"OwnerToOperator:{owner}:{operatorAddress}", value);
+
+    /// <summary>
+    /// Owner of the contract is responsible to for minting/burning 
+    /// </summary>
+    public Address Owner
     {
-        this.PersistentState.SetBool($"OwnerToOperator:{owner}:{operatorAddress}", value);
+        get => State.GetAddress(nameof(Owner));
+        private set => State.SetAddress(nameof(Owner), value);
     }
 
     /// <summary>
-    /// Constructor. Initializes the supported interfaces.
+    /// Name for non-fungible token contract
     /// </summary>
-    /// <param name="state">The smart contract state.</param>
-    public NonFungibleToken(ISmartContractState state) : base(state)
+    public string Name
     {
-        // todo: discuss callback handling and supported interface numbering with community.
-        this.SetSupportedInterfaces((uint)0x00000001, true); // (ERC165) - ISupportsInterface
-        this.SetSupportedInterfaces((uint)0x00000002, true); // (ERC721) - INonFungibleToken,
-        this.SetSupportedInterfaces((uint)0x00000003, false); // (ERC721) - INonFungibleTokenReceiver
+        get => State.GetString(nameof(Name));
+        private set => State.SetString(nameof(Name), value);
     }
 
     /// <summary>
-    /// Function to check which interfaces are supported by this contract.
+    /// Symbol for non-fungible token contract
     /// </summary>
-    /// <param name="interfaceID">Id of the interface.</param>
-    /// <returns>True if <see cref="interfaceID"/> is supported, false otherwise.</returns>
-    public bool SupportsInterface(uint interfaceID)
+    public string Symbol
+    {
+        get => State.GetString(nameof(Symbol));
+        private set => State.SetString(nameof(Symbol), value);
+    }
+
+    private string TokenURIFormat
+    {
+        get => State.GetString(nameof(TokenURIFormat));
+        set => State.SetString(nameof(TokenURIFormat), value);
+    }
+
+    private bool OwnerOnlyMinting
+    {
+        get => State.GetBool(nameof(OwnerOnlyMinting));
+        set => State.SetBool(nameof(OwnerOnlyMinting), value);
+    }
+
+    /// <summary>
+    /// The next token id which is going to be minted
+    /// </summary>
+    private ulong TokenIdCounter
     {
-        return GetSupportedInterfaces(interfaceID);
+        get => State.GetUInt64(nameof(TokenIdCounter));
+        set => State.SetUInt64(nameof(TokenIdCounter), value);
+    }
```

Comment:
I think it would be beneficial to have this public. Especially if you're looking for an old TokenId you know you own.

Reply:
There are different strategies for Minting tokens. Some contracts prefer that client specify tokenid on mint method and others prefer auto incremented id numbers by contract. In our case we use auto incremented and Mint methods already returns minted tokenId. 

Client doesn't need to and have to know what is next id for contracts designed both way. 

Any application is going to rely on TokenIdCounter is going to fail when switching to another contract has no auto incremented token id strategy. 

For example, StoreContract is able to manage tokens from different ERC721 contracts and StoreContract is going to fail that if it is assumed that all these ERC721 contracts have TokenIdCounter or auto incremented token id strategy.

Reply:
Little bit more info Minting methods, mint/burn methods are not part of ERC721 and actually this is not issue because  applications that manage ERC721 tokens only interact with ERC721 interface so naturally mint/burn methods are private to your own application.

Reply:
There is no downside to making these getters public, I think they should be public when there is no harm done. It gives flexibility for future needs that may not be thought of right now. Locking by making private makes the contract more of a black box.